File: Tree/1022. Sum of Root To Leaf Binary Numbers.py
class Solution:
    def sumRootToLeaf(self, root: Optional[TreeNode]) -> int:
        def dfs(node, prev):
            if not node:
                return 0
            prev = prev + str(node.val)
            if not node.left and not node.right:
                return int(prev, 2)
            return dfs(node.left, prev) + dfs(node.right, prev)
        return dfs(root, '')

File: Tree/1022. Sum of Root To Leaf Binary Numbers.cpp
class Solution {
public:
    void helper(TreeNode *root, int out, int &res)
    {
        if (!root) return;

        out = (out << 1) + root->val;

        if (!root->left && !root->right)
        {
            res += out;
            return;
        }

        helper(root->left, out, res);
        helper(root->right, out, res);
    }

    int sumRootToLeaf(TreeNode *root)
    {
        int res = 0;
        helper(root, 0, res);
        return res;
    }
};